092-self-healing-using-cruise-control.md

```diff
@@ -0,0 +1,300 @@
+ Self healing for DISK_FAILURE: {
+	Disk /var/lib/kafka/data-1/kafka-log2 on broker 2 failed at 2024-12-09T09:49:46Z
+ }}
+```
+
+where:
+* `state` refers to the status of the replica movement (pending/in-progress/finished etc.)
+* `triggeredSelfHealingTaskId` points to the anomalyId.
+* `triggeredTaskReason` refers to the reason why self-healing was triggered.
+
+#### Using metrics when using self-healing
+
+Cruise control provides [self-healing related metrics](https://github.com/linkedin/cruise-control/wiki/Sensors#anomalydetector-sensors) which can help users understand what kind of anomalies were detected in their cluster, when self-healing started, the rate at which the anomalies are being detected in their cluster and much more.
+These metrics can also provide a way for the users to debug self-healing related issues.
+For example, the `number_of_self_healing_started` metric can provides users, the way to check when self-healing was started for an anomaly, and the number of self-healing actions that were performed in the cluster.
+This information can be useful to at what point of time the self-healing occurred and the shape/replica assignment of the cluster was changed due to self-healing.
+The user can also use the  metric like `number_of_self_healing_failed` to know if the self-healing failed to start to understand, and then they can check the reasons in the log for the failure. One example could be load monitor was ready at that point of time.
+Cruise Control also have [metrics](https://github.com/linkedin/cruise-control/wiki/Sensors#executor-sensors) for tracking the inter/intra broker replica movement and also the ongoing replica movement rate/concurrency which can be useful to know if all the replicas movement was successful or if there were some replica movement which didn't complete.
+[Work](https://github.com/linkedin/cruise-control/pull/2268) is currently ongoing in the upstream Cruise Control project to have some metrics which can give us the self-healing start and end timestamp based upon the type of anomaly.
+To enable the metrics, the user can follow the Strimzi documentation on how we [expose the metrics](https://strimzi.io/docs/operators/latest/full/deploying.html#assembly-metrics-setup-str) through an HTTP endpoint.
+
+### Proposal for enabling self-healing in Strimzi
+
+In order to activate self-healing, users will need to set the `self.healing.enabled` property to `true` and specifying a notifier that the self-healing mechanism is going to use in the `spec.cruiseControl.config` section of their `Kafka` resource.
+If the user forgets to set the notifier property then the `KubernetesEventNotifier`(custom notifier under Strimzi org) will be used by default and the user will also get a warning in the Kafka CR regarding missing `anomaly.notifier.class` property.
+We will support most anomaly fixes however there are some fixes which we don't want self-healing to fix automatically in a Kubernetes context.
```

Comment:
Can you be specific and list what goals will be and what will not be supported instead of saying `most`?

Also, given what you say below and assuming it does not change -> please make it clear here that they will be just disabled by default while the users can enable them.